cmd/kubermatic-installer/cmd_local.go

```
@@ -0,0 +1,418 @@
+		MLASkipMinioLifecycleMgr:   true,
+		KubermaticConfiguration:    k,
+		RawKubermaticConfiguration: uk,
+		HelmValues:                 v,
+	}
+
+	if err := ms.Deploy(context.Background(), msOpts); err != nil {
+		logger.Fatalf("Failed to deploy kubermatic stack: %v", err)
+	}
+
+	kubeconfig := filepath.Join(dir, "kube-config.yaml")
+	internalKubeconfig := filepath.Join(dir, "kube-config-internal.yaml")
+	kindSeedSecret := initKindSeedSecret(kubeClient, logger, kubeconfig, internalKubeconfig)
+	ensureResource(kubeClient, logger, &kindSeedSecret)
+	kindPreset := initKindPreset(logger, internalKubeconfig)
+	ensureResource(kubeClient, logger, &kindPreset)
+	ensureResource(kubeClient, logger, &kindLocalSeed)
+	return kkpEndpoint
+}
+
+func localKindFunc(logger *logrus.Logger, opt LocalOptions) cobraFuncE {
+	return handleErrors(logger, func(cmd *cobra.Command, args []string) error {
+		dir := "./examples"
+		path, err := os.Executable()
+		if err == nil {
+			dir = filepath.Join(filepath.Dir(path), "examples")
+		}
+		kubeClient, cancel := localKind(logger, dir)
+		defer cancel()
+		kubeconfig := filepath.Join(dir, "kube-config.yaml")
+		helmClient, err := helm.NewCLI(opt.HelmBinary, kubeconfig, "", opt.HelmTimeout, logger)
+		if err != nil {
+			logger.Fatalf("Failed to create helm client: %v", err)
+		}
+		helmVersion, err := helmClient.Version()
+		if err != nil {
+			logger.Fatalf("Failed to check Helm version: %v", err)
+		}
+		if helmVersion.LessThan(MinHelmVersion) {
+			logger.Fatalf(
+				"the installer requires Helm >= %s, but detected %q as %s (use --helm-binary or $HELM_BINARY to override)",
+				MinHelmVersion,
+				opt.HelmBinary,
+				helmVersion,
+			)
+		}
+		installKubevirt(logger, dir, helmClient, opt)
+		endpoint := installKubermatic(logger, dir, kubeClient, helmClient, opt)
+		logger.Infof("KKP installed successfully, login at http://%v", endpoint)
+		return nil
+	})
+}
+
+func getLocalEndpoint(logger *logrus.Logger, opts LocalOptions) string {
```

I feel like you need to return an error, and handle it in the caller ;)

---

they are handled in the function, the caller would just do the same
https://github.com/kubermatic/kubermatic/blob/bb80ea4a228766a645a88362872cfc4dfad16c4f/cmd/kubermatic-installer/cmd_local.go#L332
https://github.com/kubermatic/kubermatic/blob/bb80ea4a228766a645a88362872cfc4dfad16c4f/cmd/kubermatic-installer/cmd_local.go#L336
https://github.com/kubermatic/kubermatic/blob/bb80ea4a228766a645a88362872cfc4dfad16c4f/cmd/kubermatic-installer/cmd_local.go#L344